### PR TITLE
Update checkedCast in C++, C# and more

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR updates checkedCast in C++, C#, PHP, Ruby and Swift.

checkedCast no longer handles any exception.

See #2323.

### CODE REVIEW

File: cpp/include/Ice/ProxyFunctions.h [hunk 1]
```diff
@@ -115,45 +115,29 @@ namespace Ice
      * @param proxy The source proxy.
      * @param facet A facet name.
      * @param context The context map for the invocation.
-     * @return A proxy with the requested type and facet, or nullopt if the target object does not have the requested
-     * facet or if the facet is not of the requested type.
+     * @return A proxy with the requested type and facet, or nullopt or if the target facet is not of the requested
```
Comment (pepone) [inline cpp/include/Ice/ProxyFunctions.h:118]: `or nullopt or if the target facet...` this doesn't sound correct.

**Review by pepone — APPROVED** [review]

File: cpp/include/Ice/ProxyFunctions.h (hunk 1 above)
Comment (externl) [inline cpp/include/Ice/ProxyFunctions.h:118]:
```suggestion
     * @return A proxy with the requested type and facet, or nullopt if the target facet is not of the requested
```

**Review by externl — APPROVED** [review]

File: csharp/test/Ice/admin/AllTests.cs [hunk 2]
```diff
@@ -486,11 +514,24 @@ public static void allTests(global::Test.TestHelper helper)
                     props.Add("Ice.Admin.Facets", "TestFacet");
                     var com = factory.createCommunicator(props);
                     Ice.ObjectPrx obj = com.getAdmin();
-                    Ice.PropertiesAdminPrx pa = Ice.PropertiesAdminPrxHelper.checkedCast(obj, "Properties");
-                    test(pa == null);
-                    Ice.ProcessPrx proc = Ice.ProcessPrxHelper.checkedCast(obj, "Process");
-                    test(proc == null);
-                    com.destroy();
```
Comment (InsertCreativityHere) [inline csharp/test/Ice/admin/AllTests.cs:493]: You removed a (seemingly) unrelated `com.destroy();` call here.

**Review by InsertCreativityHere — APPROVED** [review]
Looks good, and simpler, I like it.

File: csharp/test/Ice/admin/AllTests.cs (hunk 2 above)
Comment (bernardnormier) [reply csharp/test/Ice/admin/AllTests.cs:493]: Good catch. This was accidental. Fixed.

**Pull request merged**